src/frontends/lean/definition_cmds.cpp

```diff
@@ -678,6 +680,7 @@ class proof_elaboration_task : public task<expr> {
             finalize_theorem_proof(elab, params, val, m_finfo);
             if (m_is_rfl_lemma && !is_rfl_lemma(m_final_type, val))
                 throw exception("not a rfl-lemma, even though marked as rfl");
+            if (elab.has_errors()) throw std::exception();
```

Comment:
I'm assuming that from now on, we use `std::exception()` when we want to interrupt the elaborator without reporting yet another error. Is that right?